release.py

```diff
@@ -4,129 +4,98 @@
-
+    # Set up paths
+    print("Setting up paths...")
+    abspath = os.path.abspath(__file__)
+    dname = os.path.dirname(abspath)
+    os.chdir(dname)
+
+    # Parse the command line arguments
+    print("Parsing cmdline...")
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config",     help="Sets the config file", default="config.yml")
+    parser.add_argument("--version",    help="The version to mark this release as")
+    parser.add_argument("--type",       help="Either \'candidate\' or \'release\'")
+    parser.add_argument("--candidate",  help="If --type = \'candidate\', this specifies the candidate number.  Is ignored if --type = \'release\'")
+
+    args = parser.parse_args()
+
+    # Read in configuration data from config.yml
+    print("Parsing config.yml...")
+    config = {}
+
+    with open(args.config, "r") as f:
+        try:
+            config = yaml.safe_load(f)
+            # Support some variables in the config
+            expand_config_vars(config)
+        except yaml.YAMLError as e:
+            print(e)
+            sys.exit(1)
+
+    # Checkout repo
+    print("Checking out repo: {}/{}".format(config["git"]["repo"], config["git"]["branch"]))
+    repo = git.GitRepository(config["git"]["repo"], config["git"]["branch"])
+    repo.update_repository()
+
+    # Verify version string is valid
+    print("Verifying version string...")
     if not semantic_version.validate(args.version):
-        print("Specified version is not a valid version string!")
-        return
+        print("Error: Specified version is not a valid version string!")
+        sys.exit(1)
 
+    # Form tag string according to type
+    print("Forming tag name...")
     version = semantic_version.Version(args.version)
-    if script_state is None:
-        # An existing script state overrides the commandline argument
-        if args.tag_name is not None:
-            script_state = ReleaseState(version)
-            script_state.state = ScriptState.STATE_TAG_PUSHED
-            script_state.tag_name = args.tag_name
-        else:
-            if script_state is None:
-                script_state = ReleaseState(version)
+    tag_name = ''
+    if (args.type == "candidate"):
+        # suffix RC# to version
+        if (not isinstance(args.candidate, int)) or (args.candidate < 0):
+            print ("Error: Candidate field must be a non-negative integer!")
+            sys.exit(1)
+
+        tag_name = "release_{}.{}.{}_RC{}".format(version.major, version.minor, version.build, args.candidate)
+
+        # Don't configure version_override.cmake for RC's.  They identify as the previous unstable version
+
+    elif (args.type == "release"):
+        # no suffix
+        tag_name = "release_{}.{}.{}".format(version.major, version.minor, version.build)
+
+        # Configure version_override.cmake for proper in-game version ident
+        with open(os.path.join(config["git"]["repo"], "version_override.cmake"), "a") as f:
```

Comment:
This needs to be moved out of the release branch so that RCs also get the version override.

Reply:
To my knowledge our RC's have identified in-game as an unstable build and haven't identified themselves as an RC.  Identifying as an RC at last would probably be a good idea...

Reply:
If that is currently the case then it was an oversight before. An RC build should also have this version override.